docs/rules/object-property-newline.md

```diff
@@ -0,0 +1,89 @@
+# Enforce placing object properties on separate lines (object-property-newline)
+
+While formatting preferences are very personal, a number of style guides require that object properties be placed on separate lines for better readability.
+
+Another argument in favor of this style is that it improves the readability of diffs when a property is changed:
+
+```diff
+// More readable
+ var obj = {
+     foo: "foo",
+-    bar: "bar",
++    bar: "bazz",
+     baz: "baz"
+ };
+```
+
+```diff
+// Less readable
+-var obj = { foo: "foo", bar: "bar", baz: "baz" };
++var obj = { foo: "foo", bar: "bazz", baz: "baz" };
+```
+
+## Rule Details
+
+This rule aims to maintain consistency of newlines between object properties.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint object-property-newline: "error"*/
+
+var obj = { foo: "foo", bar: "bar", baz: "baz" };
+
+var obj2 = {
+    foo: "foo", bar: "bar", baz: "baz"
+};
+
+var obj3 = {
+    foo: "foo", bar: "bar",
+    baz: "baz"
+};
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint object-property-newline: "error"*/
+
+var obj = {
+    foo: "foo",
+    bar: "bar",
+    baz: "baz"
+};
+```
+
+## Options
+
+This rule has an object option:
+
+* `"allowMultiplePropertiesPerLine"`: `true` allows all keys and values to be on the same line
+
+### "allowMultiplePropertiesPerLine"
```

Comment:
Because the example sentence and eslint comment provide exact syntax for rule configuration, you can omit the quote marks in the heading to limit the first impression of visual complexity.